lib/linter/linter.js

```diff
@@ -2340,6 +2341,23 @@ class Linter {
                 }
             }
 
+            // Stop if we've made a circular fix
+            if (recentOutputs.length === 2 && fixedResult.output === recentOutputs[0]) {
```

Comment:
Because it's expensive to compare large strings, let's check the length first to make sure it's worth taking that performance hit.
```suggestion
            if (recentOutputs.length === 2 && fixedResult.output.length === recentOutputs[0].length && fixedResult.output === recentOutputs[0]) {
```

Reply:
Is `fixedResult.output` long enough so that this is a significant performance optimization?

Reply:
Even comparing single-character strings in a loop can cause a significant performance degradation.

Reply:
to reduce performance penalty, would it make sense to move this logic inside the `if (fixedResult.fixed) {` (line 2391)?
in that case, you can check for `passNumber === MAX_AUTOFIX_PASSES` before checking autofix results